### PR TITLE
FINISHED (Uploads to Bucket now)

### PR DESCRIPTION
**Pull request by JohnFawkes**

### CODE REVIEW

**Comment by ThatNerdyPikachu** [comment]
hey, great work! will test tomorrow.
1. how is patch xml defined in install, but still in scope in post fs data?
2. version code 262 -> 260

(am on mobile can't do a review that well)

**Comment by JohnFawkes** [comment]
I'm not finished yet and I only did versioncode 262 so it would update on my device when I was flashing to test so I didn't have to flash 2 times

**Comment by JohnFawkes** [comment]
Please go over code. ive sent a test zip with these changes in it to the TG group. please test to make sure everything is working correctly

File: config.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ AUTOMOUNT=true
 PROPFILE=true
 
 # Set to true if you need post-fs-data script
-POSTFSDATA=true
+POSTFSDATA=false
```
Comment (thehappydinoa) [inline config.sh]: You disabled a whole file you added a lot to.

File: common/post-fs-data.sh [hunk 2]
```diff
@@ -1,18 +1,76 @@
...
-    sed -i 's/name="__data_rollout__SpeakEasy.OverrideUSLocaleCheckRollout__launched__" value="false"/name="__data_rollout__SpeakEasy.OverrideUSLocaleCheckRollout__launched__" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="G__enable_speakeasy_details" value="false"/name="G__enable_speakeasy_details" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="G__speak_easy_enabled" value="false"/name="G__speak_easy_enabled" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="G__speakeasy_show_privacy_tour" value="false"/name="G__speakeasy_show_privacy_tour" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="__data_rollout__SpeakEasy.SpeakEasyDetailsRollout__launched__" value="false"/name="__data_rollout__SpeakEasy.SpeakEasyDetailsRollout__launched__" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="__data_rollout__SpeakEasy.CallScreenOnPixelTwoRollout__launched__" value="false"/name="__data_rollout__SpeakEasy.CallScreenOnPixelTwoRollout__launched__" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="G__speakeasy_postcall_survey_enabled" value="false"/name="G__speakeasy_postcall_survey_enabled" value="true"/g' $DIALER_PREF_FILE
+    #DIALERPATCHES
```
Comment (thehappydinoa) [inline common/post-fs-data.sh:73]: I assume this is being worked on.

File: common/install.sh [hunk 3]
```diff
@@ -199,7 +256,21 @@ fi
...
+    patch_xml -s $DPF '/map/boolean[@name="__data_rollout__SpeakEasy.OverrideUSLocaleCheckRollout__launched__"]' "true"
+    patch_xml -s $DPF '/map/boolean[@name="G__enable_speakeasy_details"]' "true"
+    patch_xml -s $DPF '/map/boolean[@name="G__speak_easy_enabled"]' "true"
+    patch_xml -s $DPF '/map/boolean[@name="G__speakeasy_show_privacy_tour"]' "true"
+    patch_xml -s $DPF '/map/boolean[@name="__data_rollout__SpeakEasy.SpeakEasyDetailsRollout__launched__"]' "true"
+    patch_xml -s $DPF '/map/boolean[@name="__data_rollout__SpeakEasy.CallScreenOnPixelTwoRollout__launched__"]' "true"
+    patch_xml -s $DPF '/map/boolean[@name="G__speakeasy_postcall_survey_enabled"]' "true"
+  fi
+fi
+  if [ "$SLIM" == "false" ]; then
```
Comment (thehappydinoa) [inline common/install.sh]: You need to fix that if/fi indentation. I think you added an extra one on line 288.

File: config.sh [hunk 4]
```diff
@@ -88,7 +88,8 @@ print_modname() {
   ui_print "    *<name>*"
   ui_print "    *******************************************"
   ui_print "    *<version>*"
-  ui_print "    *<author>*"
+  ui_print "    *    Joey Huab, Aidan Holland, Pika       *" 
+  ui_print "    *    John Fawkes, Laster K. (lazerl0rd)   *"
```
Comment (thehappydinoa) [inline config.sh]: This looks a lot better.

File: module.prop [hunk 5]
```diff
@@ -1,7 +1,7 @@
 id=Pix3lify
 name=Pix3lify
-version=v2.5.2
-versionCode=252
+version=v2.6
+versionCode=260
 author=Joey Huab, Aidan Holland, Pika, JohnFawkes, Laster K
 description=Ports Pixel 3 exclusive features; Pixel Blue theme accent, Alarms, Notification sounds, Ringtones, UI audio; enables Pixel Wallpapers, enables Google Call Screening, enables Night Light (device dependent), Daydream VR support, adds Device Gestures to Settings, enables Camera2 API and EIS support (varies per device)
 support=https://forum.xda-developers.com/apps/magisk/module-pixel-2-experience-t3757137
```
Comment (thehappydinoa) [inline module.prop]: Should we keep XDA as the support link?

**Review by thehappydinoa — CHANGES_REQUESTED** [review]
Overall looks solid, but I think you need to pull from `master` I made some bug fixes and removed the `bootanimation.zip` from `/system`. Also make sure to edit the `README.md` about your role!

File: config.sh (hunk 1 above)
Comment (JohnFawkes) [reply config.sh]: Oops

File: common/post-fs-data.sh (hunk 2 above)
Comment (JohnFawkes) [reply common/post-fs-data.sh:73]: Nope. This is fine. The install.sh copy the patch_xml function copys the patchs to #DIALERPATCHES in post-fs-data

File: module.prop (hunk 5 above)
Comment (JohnFawkes) [reply module.prop]: Nah we should change that to TG link

File: common/post-fs-data.sh (hunk 2 above)
Comment (JohnFawkes) [reply common/post-fs-data.sh:73]: case $2 in
    *dialer_phenotype_flags*.xml)  sed -i "/#DIALERPATCHES/a\          patch_xml $1 \$MODPATH/\ '$3' \"$4\"" $INSTALLER/common/post-fs-data.sh; VAR1=boolean; VAR2=string; VAR3=long;;

File: module.prop (hunk 5 above)
Comment (thehappydinoa) [reply module.prop]: That's what I was going to suggest.

File: common/post-fs-data.sh (hunk 2 above)
Comment (thehappydinoa) [reply common/post-fs-data.sh:73]: Ah cool

File: common/install.sh [hunk 6]
```diff
@@ -119,7 +119,7 @@ cp_ch -n $INSTALLER/pix3lify.sh $UNITY$BINPATH/pix3lify
 log_handler "Using $BINPATH."
 sed -i -e "s|<MAGISK>|$MAGISK|" -e "s|<CACHELOC>|$CACHELOC|" -e "s|<BINPATH>|$BINPATH|" -e "s|<MODVERSION>|$(grep_prop versionCode $INSTALLER/module.prop)|" -e "s|<MODID>|$MODID|" $UNITY$BINPATH/icewizard
 if $MAGISK; then
-sed -i -e "s|<PROP>|$(echo $PROP)|" -e "s|<MODPATH>|/sbin/.magisk/img/Pix3lify|" -e "s|<MODPROP>|$(echo $MOD_VER)|" -e "s|<MOUNTPATH>|/sbin/.core/img|" $UNITY$BINPATH/pix3lify
+sed -i -e "s|<PROP>|$(echo $PROP)|" -e "s|<MODPATH>|/sbin/.magisk/img/Pix3lify|" -e "s|<MODPROP>|$(echo $MOD_VER)|" -e "s|<MOUNTPATH>|/sbin/.magisk/img|" $UNITY$BINPATH/pix3lify
```
Comment (lzlrd) [inline common/install.sh]: Won’t this drop support for older Magisk versions?

Comment (JohnFawkes) [reply common/install.sh]: Older than 18 yes. I'll see if i can change it to support older

Comment (lzlrd) [reply common/install.sh]: This module (https://github.com/null4n/vold-posix) has an example of how to detect different versions of Magisk, but you probably got an idea anyways :P.

Comment (JohnFawkes) [reply common/install.sh]: yep i already figured it out just havent pushed the changes yet. 

MAGISK_VERSIONCODE=$(echo $(get_file_value /data/adb/magisk/util_functions.sh "MAGISK_VERSIONCODE=") | sed 's|-.*||')

get_file_value() {
	if [ -f "$1" ]; then
		cat $1 | grep $2 | sed "s|.*${2}||" | sed 's|\"||g'
	fi
}

if $MAGISK_VERSIONCODE -ge "18000"; then 
MOUNTPATH=/sbin/.magisk/img
else
MOUNTPATH=/sbin/.core/img
fi

MODPATH=$MOUNTPATH/Pix3lify

File: README.md [hunk 7]
```diff
@@ -58,8 +58,29 @@ As a Mi A1 user, I was stuck with Stock Oreo 8.0.0 without Oreo custom ROMs when
...
 ## ⭐ Changelog
 
+### v3.0
+
+-  Update Unity fixes
+-  Introduce logging
+-  Add pix3lify terminal script to send logs
+-  Bug fixes/typos
+
+### v2.6
```
Comment (thehappydinoa) [inline README.md]: Do we just want to combine?

File: config.sh [hunk 8]
```diff
@@ -72,6 +81,44 @@ unity_custom() {
     BFOLDER="/system/media/"
     BZIP="bootanimation.zip"
   fi
+  MODTITLE=$(grep_prop name $INSTALLER/module.prop)
+  VER=$(grep_prop version $INSTALLER/module.prop)
+	AUTHOR=$(grep_prop author $INSTALLER/module.prop)
```
Comment (thehappydinoa) [inline config.sh]: Spacing is weird here

File: pix3lify.sh [hunk 9]
```diff
@@ -0,0 +1,436 @@
...
+}
+
+# Log files will be uploaded to termbin.com
+upload_logs() {
+	$BBok && {
+		test_connection
+		[ $? -ne 0 ] && exit
+		logup=none;
+		echo "Uploading logs"
+		[ -s $XZLOG ] && logup=$(cat $XZLOG | curl -i -F "Pix3lify_logs.tar.xz"  http://logs.pix3lify.com/submit)
```
Comment (thehappydinoa) [inline pix3lify.sh]: This has to be `curl -i -F "logs=@Pix3lify_logs.tar.xz"  http://logs.pix3lify.com/submit`. I will work on the `.tar.xz` support.

File: pix3lify.sh [hunk 10]
```diff
@@ -0,0 +1,436 @@
...
+	cd $CACHELOC
+	tar -zcvf Pix3lify_logs.tar.gz Pix3lify_logs >> $ALOG 2>&1
+
+  	# Copy package to internal storage
+	mv -f $CACHELOC/Pix3lify_logs.tar.gz $SDCARD >> $ALOG 2>&1
+
+if  [ -e $SDCARD/Pix3lify_logs.tar.gz ]; then 
+  log_print "Pix3lify_logs.tar.gz Created Successfully. Please Upload to Telegram, and tag @JohnFawkes"
+else
+  log_print "Zip File Not Created. Error in Script"
```
Comment (thehappydinoa) [inline pix3lify.sh]: Maybe we change this to `"Archive Not Created"`

**Review by thehappydinoa — CHANGES_REQUESTED** [review]
Looks pretty solid, just a few things here and there.

File: README.md (hunk 7 above)
Comment (JohnFawkes) [reply README.md]: yea thats fine

File: config.sh (hunk 8 above)
Comment (JohnFawkes) [reply config.sh]: fixed

File: pix3lify.sh (hunk 9 above)
Comment (JohnFawkes) [reply pix3lify.sh]: fixed

File: pix3lify.sh (hunk 10 above)
Comment (JohnFawkes) [reply pix3lify.sh]: yep i can change that.

**Comment by JohnFawkes** [comment]
this should be able to be merged today. just a few more things to fix and then tar.xz support on server and we should be good

**Comment by thehappydinoa** [comment]
Perfect!

File: pix3lify.sh (hunk 10 above)
Comment (JohnFawkes) [reply pix3lify.sh]: glad you noticed this because i need to change all the .gz there to .xz

**Comment by thehappydinoa** [comment]
The server is all set

**Comment by JohnFawkes** [comment]
> The server is all set

Awesome! I'm starting to get a migraine so I'm gunna take some Tylenol and lay down for a little bit to get rid of it then ill finish the last couple of bugs and then push. Should I close this PR out then make a brand new one so all the changes are in one commit and easier to follow?

**Comment by lzlrd** [comment]
@JohnFawkes you can just change the title and description of this merge then squash merge (it's an option in desktop view of this page next to "close pr/comment"). It'll put it under one commit with the title and description of this PR.

**Comment by thehappydinoa** [comment]
> @JohnFawkes you can just change the title and description of this merge then squash merge (it's an option in desktop view of this page next to "close pr/comment").

I think this might be best, but it might be nice to have new reviews and things.

File: pix3lify.sh (hunk 9 above)
Comment (thehappydinoa) [reply pix3lify.sh]: Actually, drop the `-i` you just need the key that my server responses with.

Comment (JohnFawkes) [reply pix3lify.sh]: Sounds good. I've got all the changes, just trying to test. Magisk is screwing up on me. Got a black screen during install of a different module and now I lost all my installed modules. So trying to get magisk back working properly so I can test

Comment (thehappydinoa) [reply pix3lify.sh]: Ya I just had something similar happen. Had to factory reset.

File: common/install.sh [hunk 11]
```diff
@@ -49,170 +111,197 @@ case $(basename $ZIP) in
...
-    ui_print "   Pix3lify has been known to not work and cause issues on devices running OxygenOS!"
-  else
-    ui_print "   Pix3lify is only for non-Google devices!"
-  fi
-  ui_print "   DO YOU WANT TO IGNORE OUR WARNINGS AND RISK A BOOTLOOP?"
-  ui_print "   Vol Up = Yes, Vol Down = No"
+  log_print "   Pix3lify is only for non-Google devices!"
+  log_print "   DO YOU WANT TO IGNORE OUR WARNINGS AND RISK A BOOTLOOP?"
+  log_print "   Vol Up = Yes, Vol Down = No"
   if $FUNCTION; then
```
Comment (thehappydinoa) [inline common/install.sh:93]: Referenced before definition.

**Comment by JohnFawkes** [comment]
this is finally all done. unity 3.1, Full logging with tar.xz creation. only thing left is server upload which i havent figured out. in icewizard i just have the user send me the tar.xz directly.

File: config.sh [hunk 12]
```diff
@@ -72,9 +82,58 @@ unity_custom() {
     BFOLDER="/system/media/"
     BZIP="bootanimation.zip"
   fi
-}
+  MODTITLE=$(grep_prop name $INSTALLER/module.prop)
+  VER=$(grep_prop version $INSTALLER/module.prop)
+	AUTHOR=$(grep_prop author $INSTALLER/module.prop)
```
Comment (thehappydinoa) [inline config.sh]: Is the spacing weird here?

File: pix3lify.sh [hunk 13]
```diff
@@ -0,0 +1,448 @@
...
+# > Rather than editing the default vars above.
+
+#Log Functions
+# Saves the previous log (if available) and creates a new one
+log_start() {
+if [ -f "$LOG" ]; then
+	mv -f $LOG $OLDLOG
+fi
+touch $LOG
+echo " " >> $LOG 2>&1
```
Comment (thehappydinoa) [inline pix3lify.sh:284]: We have multiple headers?

File: config.sh (hunk 12 above)
Comment (JohnFawkes) [reply config.sh]: i dont see why. its not like that in my editor. ill check again though

File: pix3lify.sh (hunk 13 above)
Comment (JohnFawkes) [reply pix3lify.sh:284]: headers?

File: config.sh (hunk 12 above)
Comment (thehappydinoa) [reply config.sh]: Oh that's my bad just looked funky

File: pix3lify.sh (hunk 13 above)
Comment (thehappydinoa) [reply pix3lify.sh:284]: Like logos and ascii stuff

Comment (JohnFawkes) [reply pix3lify.sh:284]: yes we do.

**Review by joeyhuab — APPROVED** [review]
There's honestly too much going on with the changes and my head hurts trying to understand everything new so I'm just gonna give this the green light.

**Comment by JohnFawkes** [comment]
@stallix there's still a few issues to fix. The curl stuff needs fixed and I think one other issue I can't remember at the moment without looking at it. The core functionality is there though and the tar.xz gets created, it just doesn't upload to our server yet

**Comment by JohnFawkes** [comment]
Time for 2.6 release

**Pull request merged**